### PR TITLE
Able to clear all comments from datastore

### PR DESCRIPTION
**Pull request by margaretchan**
Primary Reviewer: Derrill
Secondary Reviewer: Sam

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        Query query = new Query("Comment");
+        PreparedQuery results = datastore.prepare(query);
+        
+        // iterate through query and remove all keys
+        for (Entity entity : results.asIterable()) {
+            Key commentKey = entity.getKey();
+            datastore.delete(commentKey);
+        }
+
+        response.getWriter().println("");
```
Comment (derrillhd) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:50]: Printing a newline is unnecessary.  remove.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.List;
+import java.util.Date;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for deleting comment data */
+@WebServlet("/delete-data")
```
Comment (derrillhd) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:35]: so I kind of ignored the REST stuff, but I'm going to perhaps start enforcing it.  In your other servlet use a RESTful API design such that the URI is "/comments/*".

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+public class DeleteDataServlet extends HttpServlet {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (derrillhd) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:39]: Please read the comments I gave dante about RESTful design and the DELETE method:
https://github.com/dallen68/GoogleStep/pull/6/files

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        
+        // iterate through query and remove all keys
+        for (Entity entity : results.asIterable()) {
+            Key commentKey = entity.getKey();
+            datastore.delete(commentKey);
+        }
+
+        response.getWriter().println("");
+    }
+}
```
Comment (derrillhd) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:52]: super nit:  new line at end of file fits style guide

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 1 above)
Comment (margaretchan) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:50]: removed.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 4 above)
Comment (margaretchan) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:52]: added new line

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Query query = new Query("Comment");
+        PreparedQuery results = datastore.prepare(query);
+        
+        // iterate through query and remove all keys
+        for (Entity entity : results.asIterable()) {
+            Key commentKey = entity.getKey();
+            datastore.delete(commentKey);
```
[the remark this answers is not in the document]
Comment (margaretchan) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:47]: I can access it! Would using a GQL query make the deletes atomic?

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 3 above)
Comment (margaretchan) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:39]: gotcha, doDelete would add more functionality as well

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 2 above)
Comment (margaretchan) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:35]: changed to /comments/

**Pull request merged**